nucypher/network/server.py

```diff
@@ -217,7 +217,13 @@ def reencrypt():
         capsules_to_process = list()
         for capsule, lingo in packets:
             # raises an exception or continues
-            evaluate_conditions_for_ursula(lingo=lingo, providers=providers, context=context)
+            error = evaluate_conditions(
+                lingo=lingo, providers=providers, context=context
+            )
+            if error:
+                # error cases
+                return Response(*error)
+
```

Comment:
This loop can be bypassed using a logical branch if there are no conditions in the `ReencryptionRequest`.

Reply:
I don't think this is that simple, because ReeencryptionRequest.conditions is a list with null values if there are no conditions i.e.`reenc_request.conditions` can be [null, null, null]` which gets converted to a list of `[None, None, None]` via `json.loads(...)`.

Do we go through the entire list check whether all values are `None` - in which case we are basically the same loop that we do now, just earlier. in the method? Can no conditions and conditions be mixed i.e. [None, <ConditionLingo_object>, None]? If not i.e. if the first element is `None` then all the others are, then we can just check the first entry for None - that would be easy if that premise holds...? Unsure about this.

In any case, seems like some nuance here. Left this for now, but interested to get your thoughts.

Reply:
Resolved via https://github.com/nucypher/nucypher/pull/3007/commits/f893aae94ec9f2deafae65df052b48378a805e79